src/libcollections/vec.rs

```diff
@@ -838,7 +838,11 @@ impl<T> Vec<T> {
         self.dedup_by(|a, b| key(a) == key(b))
     }
 
-    /// Removes consecutive elements in the vector that resolve to the same key.
+    /// Removes consecutive elements in the vector that compare equal according to a predicate.
```

Comment:
Could you remove `that compare equal`? It's a bit awkward here, and this is the summary line.